tls/s2n_handshake_transcript.c

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "tls/s2n_connection.h"
+#include "tls/s2n_tls.h"
+#include "tls/s2n_tls13_handshake.h"
+
+#include "stuffer/s2n_stuffer.h"
+
+#include "utils/s2n_blob.h"
+
+/* Length of the synthetic message header */
+#define MESSAGE_HASH_HEADER_LENGTH  4
+
+/* this hook runs after hashes are updated */
+int s2n_conn_post_handshake_hashes_update(struct s2n_connection *conn)
+{
```

Comment:
```
notnull_check(conn);
```

Reply:
Added the check in all functions in this file for the passed arguments.